tests/unit/options.js

```diff
@@ -2932,3 +2932,27 @@ exports.module.declarationRestrictions = function( test ) {
 
   test.done();
 };
+
+exports["gh-2436"] = function( test ) {
+  // see gh-2436. Previously jshint directives after a function declaration were ignored.
+  TestRun(test)
+    .addError(12, "This function has too many parameters. (2)")
+    .test([
+      "function x() {}",
+      "/*jshint evil: true */", // <-- previously ignored
+      "eval();",
+      "// jshint maxparams: 2",
+      "function Foo () {}",
+      "// jshint maxparams: 9", // <-- previously ignored
+      "Foo.prototype.bar = function (a, b, c, d, e) {",
+      "  return a + b + c + d + e;",
+      "};",
+      "var z = () => foo;",
+      "// jshint maxparams: 1",
+      "function y(a, b) {}",
+      "// jshint evil: false",
+      "eval()/*jshint evil: true */;"
```

Comment:
```
✖ gh-2436
Assertion Message: 
  TestRun: [bold]{undefined}
  [yellow]{Errors thrown by JSHINT, but not defined in test run}
  [bold]{Line 14, Char 1} eval can be harmful.
```

I think here the warning is correct (`evil: true` is set after `eval()`), but this would be a breaking change :disappointed: